### PR TITLE
Correct JavaScript/TypeScript typings

### PR DESCRIPTION
**Pull request by robin-da**
We use the correct JavaScript/TypeScript typings, where not already the case.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/source/app-dev/app-arch.rst [hunk 1]
```diff
@@ -81,13 +81,13 @@ your project, e.g. ``yarn add @daml/react``. Please explore the ``create-daml-ap
 to see the usage of these libraries.
 
 To make your life easy when interacting with the ledger, the DAML assistant can generate
-corresponding typescript data definitions for the data types declared in the deployed DAR.
+corresponding TypeScript data definitions for the data types declared in the deployed DAR.
```
Comment (hurryabit) [inline docs/source/app-dev/app-arch.rst:84]: The codegen does not only generate types but also JavaScript objects containing metadata about the types and templates in the DALF. We should mention that here.

File: docs/source/app-dev/app-arch.rst [hunk 2]
```diff
@@ -81,13 +81,13 @@ your project, e.g. ``yarn add @daml/react``. Please explore the ``create-daml-ap
...
 
 .. code-block:: bash
 
   daml codegen ts .daml/dist/<your-project-name.dar> -o daml.js
 
-This command will generate a typescript library for each DALF in you DAR.
+This command will generate a TypeScript library for each DALF in you DAR.
```
Comment (hurryabit) [inline docs/source/app-dev/app-arch.rst:90]: Is generates a JavaScript library that also contains TypeScript typing for the generated JavaScript.

File: docs/source/app-dev/app-arch.rst (hunk 1 above)
Comment (robin-da) [reply docs/source/app-dev/app-arch.rst:84]: I'll update the description of daml2js in the next PR. This one is only about the typings.

Comment (robin-da) [reply docs/source/app-dev/app-arch.rst:84]: I'm just working on the next PR, but I wouldn't mention the metadata here. This chapter was ment to be as simple an accesible as possible.

Comment (hurryabit) [reply docs/source/app-dev/app-arch.rst:84]: I'm not sure there's a way around mentioning the metadata. The JavaScript part of the generated code doesn't contain anything bit metadata. You don't need to go into any sort of details, just say something like "the generated JavaScript contains metadata about the types and templates in the DAML package" and that's it.

File: docs/source/daml2js/index.rst [hunk 3]
```diff
@@ -35,7 +35,7 @@ Here's a complete example that generates TypeScript from a project produced from
 
 - On execution of these commands:
 
-  - The directory ``my-proj/daml2js`` contains generated TypeScript and Javascript artifacts;
+  - The directory ``my-proj/daml2js`` contains generated TypeScript and JavaScript artifacts;
```
Comment (shayne-fletcher) [inline docs/source/daml2js/index.rst:38]: I think this PR needs rebasing. These errors I believe aren't present on master.

Comment (robin-da) [reply docs/source/daml2js/index.rst]: Has been rebased.

**Pull request merged**